docs/guides/analytics/index.md

```diff
@@ -0,0 +1,815 @@
+ * @return float|null
+ */
+protected function calculate_sample($sampleid, \core_analytics\analysable $analysable, $starttime = false, $endtime = false);
+```
+
+You may wish to add to the list of **actions** that will be offered to the recipient of an insight:
+
+```php
+/**
+ * Suggested actions for a user.
+ *
+ * @param \core_analytics\prediction $prediction
+ * @param bool $includedetailsaction
+ * @param bool $isinsightuser
+ * @return \core_analytics\prediction_action[]
+ */
+public function prediction_actions(\core_analytics\prediction $prediction, $includedetailsaction = false,
+        $isinsightuser = false)
+```
+
+You may override the users who will receive **insights notifications**:
+
+```php
+/**
+ * Returns the list of users that will receive insights notifications.
+ *
+ * Feel free to overwrite if you need to but keep in mind that moodle/analytics:listinsights
+ * or moodle/analytics:listowninsights capability is required to access the list of insights.
+ *
+ * @param \context $context
+ * @return array
+ */
+public function get_insights_users(\context $context)
+```
+
+You can implement a `always_update_analysis_time()` method so analysable elements' `timeanalysed` is only updated when analysable elements have been successfully evaluated. It is useful for lightweight targets.
+
+```php
+/**
+ * Update the last analysis time on analysable processed or always.
+ *
+ * If you overwrite this method to return false the last analysis time
+ * will only be recorded in DB when the element successfully analysed. You can
+ * safely return false for lightweight targets.
+ *
+ * @return bool
+ */
+public function always_update_analysis_time()
+```
+
+If you choose to implement `ignored_predicted_classes`, it must be public:
+
+```php
+
+/**
+ * Returns the predicted classes that will be ignored.
+ *
+ * @return array
+ */
+public function ignored_predicted_classes()
+```
+
+You can override the default message provided in the insight with `get_insight_subject()`:
+
+```php
+/**
+ * The insight notification subject.
+ *
+ * This is just a default message, you should overwrite it for a custom insight message.
+ *
+ * @param  int $modelid
+ * @param  \context $context
+ * @return string
+ */
+public function get_insight_subject(int $modelid, \context $context)
+```
+
+You can also override the URL to the insight with get_insight_context_url():
+
+```php
+/**
+ * URL to the insight.
+ *
+ * @param  int $modelid
+ * @param  \context $context
+ * @return \moodle_url
+ */
+public function get_insight_context_url($modelid, $context)
+```
+
+#### Analysis interval (core_analytics\local\time_splitting\base)
+
+Analysis intervals are used to define when the analytics API will train the predictions processor and when it will generate predictions. As explained above in the [Analysis intervals](./#analysis-intervals) documentation, they define time ranges based on analysable elements start and end timestamps.
```

Comment:
The anchor link is not working... Probably removing "./" will fix it

Reply:
Thanks